### PR TITLE
[#3031] Update mautrix-signal to the go version (signalgo merged to signal)

### PR DESCRIPTION
**Pull request by pm-McFly**
As mentioned in issue https://github.com/spantaleev/matrix-docker-ansible-deploy/issues/3031, mautrix-signal is about to be deprecated.

After testing an installation with the mautrix-signal enabled I was surprised to not even be able to complete a login or a register action.

This PR aims to provide the configuration and documentations files for mautrix-signalgo bridge.

Please let me know if I forgot to update anything.

Kindly,

### CODE REVIEW

**Comment by longregen** [comment]
This is awesome, thank you!
I'd like to add that right now mautrix-signal is unusable because of this: https://gitlab.com/signald/signald/-/issues/379

File: roles/custom/matrix-bridge-mautrix-signalgo/defaults/main.yml [hunk 1]
```diff
@@ -0,0 +1,159 @@
+---
+# mautrix-signalgo is a Matrix <-> signalgo bridge
+# Project source code URL: https://github.com/mautrix/signalgo
+
+matrix_mautrix_signalgo_enabled: true
```
Comment (longregen) [inline roles/custom/matrix-bridge-mautrix-signalgo/defaults/main.yml:5]: This should be false maybe?

File: roles/custom/matrix-bridge-mautrix-signalgo/templates/config.yaml.j2 [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+
+# Application service host/registration related details.
+# Changing these values requires regeneration of the registration.
+appservice:
+    # The address that the homeserver can use to connect to this appservice.
+    address: {{ matrix_mautrix_signalgo_appservice_address | to_json }}
+
+    # The hostname and port where this appservice should listen.
+    hostname: 0.0.0.0
+    port: 8080
```
Comment (longregen) [inline roles/custom/matrix-bridge-mautrix-signalgo/templates/config.yaml.j2:36]: This is a common port -- maybe change it for something else?

File: roles/custom/matrix-bridge-mautrix-signalgo/defaults/main.yml (hunk 1 above)
Comment (longregen) [reply roles/custom/matrix-bridge-mautrix-signalgo/defaults/main.yml:5]: Oh; sounds good; I remember noticing this in the past; thank you!

File: roles/custom/matrix-bridge-mautrix-signalgo/templates/config.yaml.j2 (hunk 2 above)
Comment (longregen) [reply roles/custom/matrix-bridge-mautrix-signalgo/templates/config.yaml.j2:36]: Oh that makes a lot of sense, thank you!!

File: group_vars/matrix_servers [hunk 3]
```diff
@@ -1090,6 +1096,51 @@ matrix_mautrix_signal_daemon_container_image_self_build: "{{ matrix_architecture
...
+matrix_mautrix_signalgo_login_shared_secret: "{{ matrix_synapse_ext_password_provider_shared_secret_auth_shared_secret if matrix_synapse_ext_password_provider_shared_secret_auth_enabled else '' }}"
+
+# People using an external Prometheus server will need to toggle all of these to be able to consume metrics remotely:
+# - `matrix_mautrix_signalgo_metrics_enabled`
+# - `matrix_mautrix_signalgo_proxying_metrics_enabled`
+# - `matrix_nginx_proxy_proxy_matrix_metrics_enabled`
+matrix_mautrix_signalgo_metrics_enabled: "{{ prometheus_enabled }}"
+
+# Postgres is the default, except if not using internal Postgres server
+matrix_mautrix_signalgo_database_engine: "{{ 'postgres' if devture_postgres_enabled else 'sqlite' }}"
```
Comment (longregen) [inline group_vars/matrix_servers:1134]: This forces the engine to be postgres if managed postgres is present, which sounds reasonable. But maybe should respect [these lines](https://github.com/spantaleev/matrix-docker-ansible-deploy/blob/173286470cc44523df878451d8efb1772afef75a/roles/custom/matrix-bridge-mautrix-signalgo/defaults/main.yml#L68-L74)?

**Review by longregen — APPROVED** [review]
Everything looks like it's working, but I'm getting this issue that seems to be some sort of encoding/decoding problem going on when trying to save it to the database. I'll try to debug it tomorrow. 

```
Dec 15 12:56:21 matrix matrix-mautrix-signalgo[69090]: Dec 15, 2023 12:56:21 DBG Message received from ffffffff-0000-0000-b1aa-d0b5eb343ee2 (group: <nil>) user_id=@user:example.com
Dec 15 12:56:21 matrix matrix-mautrix-signalgo[69090]: Dec 15, 2023 12:56:21 DBG fetchContactThenTryAndUpdateWithProfile: updating existing contact for uuid: ffffffff-0000-0000-b1aa-d0b5eb343ee2 component=signalmeow
Dec 15 12:56:21 matrix matrix-mautrix-signalgo[69090]: Dec 15, 2023 12:56:21 INF profileKey is nil component=signalmeow
Dec 15 12:56:21 matrix matrix-mautrix-signalgo[69090]: Dec 15, 2023 12:56:21 ERR fetchContactThenTryAndUpdateWithProfile: error retrieving profile for uuid: ffffffff-0000-0000-b1aa-d0b5eb343ee2 error="profile key not found" component=signalmeow
Dec 15 12:56:21 matrix matrix-mautrix-signalgo[69090]: Dec 15, 2023 12:56:21 DBG Waiting for message chat_id=ffffffff-0000-0000-b1aa-d0b5eb343ee2
Dec 15 12:56:21 matrix matrix-mautrix-signalgo[69090]: Dec 15, 2023 12:56:21 DBG StoreContactDetailsAsContact: creating new contact for uuid: aaaaaaaa-1111-1111-9a02-bffc54960129 component=signalmeow
Dec 15 12:56:21 matrix matrix-mautrix-signalgo[69090]: Dec 15, 2023 12:56:21 DBG StoreContactDetailsAsContact: no avatar found for uuid: aaaaaaaa-1111-1111-9a02-bffc54960129 component=signalmeow
Dec 15 12:56:21 matrix matrix-mautrix-signalgo[69090]: Dec 15, 2023 12:56:21 DBG StoreContactDetailsAsContact: storing contact for uuid: aaaaaaaa-1111-1111-9a02-bffc54960129 component=signalmeow
Dec 15 12:56:21 matrix matrix-mautrix-signalgo[69090]: Dec 15, 2023 12:56:21 ERR StoreContactDetailsAsContact: error storing contact error="pq: invalid byte sequence for encoding \"UTF8\": 0xb9" component=signalmeow
Dec 15 12:56:21 matrix matrix-mautrix-signalgo[69090]: Dec 15, 2023 12:56:21 ERR StoreContactDetailsAsContact error error="pq: invalid byte sequence for encoding \"UTF8\": 0xb9" component=signalmeow
```

**Comment by onestacked** [comment]
As far as I know [mautrix-signalgo](https://github.com/mautrix/signalgo) is intended to be merged into [mautrix-signal](https://github.com/mautrix/signal) once it is stable, so I'm not sure how this should be handled by the playbook.

**Comment by spantaleev** [comment]
If mautrix-signalgo would be replacing mautrix-signal (very) soon, I suppose the best path forward would be to avoid the confusion of having 2 roles that would ultimately do the same thing, and instead just wait for it to happen.

Given that they'd be merging into mautrix-signal, I suppose they will also provide a smooth migration path for users of the existing bridge, so hopping on that train would be best.

Perhaps @tulir or @smweber could tell us more

**Comment by longregen** [comment]
> If mautrix-signalgo would be replacing mautrix-signal (very) soon, I suppose the best path forward would be to avoid the confusion of having 2 roles that would ultimately do the same thing, and instead just wait for it to happen.

This merge just landed. I think this PR should be modified to be compatible with current configurations without the "go" in "signalgo".

**Comment by longregen** [comment]
> Maybe an issue in the PG configuration in the filed of a table?

According to this message, https://matrix.to/#/!mautrix-signal-v6:maunium.net/$8dzs8sw6Ypi7P41CaA2poAlMs1J_ooLAMBytcTwFZT0?via=maunium.net&via=matrix.org&via=ahouansou.cz the author has not tested it thoroughly with synapse and/or postgresql. I believe it's probably an encoding error; the bytes in the log correspond to invalid values for utf-8.

**Comment by JokerGermany** [comment]
It looks like not all features of the python bridge are in the go bridge. 
Do you really think a simple update should be done now?
In my Opinion there should be the option to update, but every users should do this manual at the moment... 
Like a Tag or something.

**Comment by spantaleev** [comment]
Is the old mautrix-signal (Python) bridge completely unusable for everyone already? Is there any hope for it receiving some fix which restores operation?

**Comment by JokerGermany** [comment]
If I am correctly informed you can't link new devices at the moment:
https://gitlab.com/signald/signald/-/issues/379

=> For new Users or users who relink their devices it should be unusable.

**Comment by longregen** [comment]
> Is the old mautrix-signal (Python) bridge completely unusable for everyone already? Is there any hope for it receiving some fix which restores operation?

I used the bridge every day for ~1 year. It suddenly stopped working about a week ago, so I'm not sure it's only new users.

I believe the authors will not patch it (the CI that creates the docker images already is pushing their :latest tag pointing to the go version)

BTW thank you for this playbook!

**Comment by onicosta** [comment]
> > Is the old mautrix-signal (Python) bridge completely unusable for everyone already? Is there any hope for it receiving some fix which restores operation?
> 
> I used the bridge every day for ~1 year. It suddenly stopped working about a week ago, so I'm not sure it's only new users.
> 
> I believe the authors will not patch it (the CI that creates the docker images already is pushing their :latest tag pointing to the go version)
> 
> BTW thank you for this playbook!

This was my experience, as well. The bridge stopped working a week ago somewhere.

**Comment by Jaffex** [comment]
I switched to this branch a few days ago and must say it works pretty well!

Migration went smooth.

Had to unlink and relink my devices twice before everything worked and room names get lost until a contact writes their first message to you after migrating, but those are kinks of the new signal bridge I guess, nothing to do with this playbook.

Can't wait to see it merged to main!

**Comment by Jaffex** [comment]
> Did you migrate from the Python code?

Yes, I had the python version before, but a problem with linked devices forced me to relink my bridge to my signal account.

Since that does not work anymore in the current state of the python bridge, I decided to give this branch a spin.

**Comment by mochman** [comment]
This fails on my server since it's an AMD64 server.

`myinstance matrix-mautrix-signal[1152020]: WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested`

It looks like you hardcode the architecture on to the docker container version in main.yaml

https://github.com/spantaleev/matrix-docker-ansible-deploy/blob/c873516cb6281feb81c540acc8f9e7ef94e7d496/roles/custom/matrix-bridge-mautrix-signal/defaults/main.yml#L12

I was able to get it to work on my server by putting 
`matrix_mautrix_signal_version: "fb18dcebcf611375ef4887041bc3b79008246178-{{ matrix_architecture }}"
`  in my `inventory/host_vars/SERVER/vars.yml`

This will at least help it work with amd64 & arm64 architectures since it looks like those are the 2 [docker images](https://mau.dev/mautrix/signal/container_registry) that are created.

**Comment by mochman** [comment]
> For the time being it seems that the docker images are not versioned anymore... Maybe this will change in the futur and fix this "hard-coding" need.
> 
> I have also updated the image version to latest (with successful tests/build) fix revision.
> 
> If you want you can use `latest-{{ matrix_architecture }}` in your `inventory` to use the most up to date released version

Thanks.  Yeah, hopefully they start adding version tags to the git repo.  Until then though, just using `matrix_mautrix_signal_version: "latest"` without the arch works fine.  It pulls the correct image.

**Comment by dhopfm** [comment]
I'm not sure that the new implementation is ready for the mass-migration of the user base which the recent merge into `master` entails. Looking at [mautrix/signal](https://github.com/mautrix/signal/commits/main/) shows that the code base is still undergoing substantial changes including the database layout and associated migrations. At this stage there's a good chance of ending up with a half-migrated DB.

**Comment by natecovington** [comment]
I recently tried updating my Ansible Playbook and ran into this error with signal:

```
Mar 18 17:42:14 poseidon systemd[1]: matrix-mautrix-signal.service: Failed with result 'exit-code'.
Mar 18 17:42:44 poseidon systemd[1]: matrix-mautrix-signal.service: Scheduled restart job, restart counter is at 198.
Mar 18 17:42:44 poseidon systemd[1]: Stopped Matrix Mautrix Signal bridge.
Mar 18 17:42:44 poseidon systemd[1]: Starting Matrix Mautrix Signal bridge...
Mar 18 17:42:44 poseidon matrix-mautrix-signal[77212]: 5c809edb14c6e52963ba3fb6dbb216a5c8a23478d0f0eb40c4e6e03b149f152d
Mar 18 17:42:44 poseidon systemd[1]: Started Matrix Mautrix Signal bridge.
Mar 18 17:42:45 poseidon matrix-mautrix-signal[77231]: Migrating legacy log config
Mar 18 17:42:45 poseidon matrix-mautrix-signal[77231]: Mar 18, 2024 21:42:45 FTL Failed to initialize database error="failed to run upgrade v11->v12: please upgrade to mautrix-signal v0.4.3 before upgrading to a newer version" db_section=main
Mar 18 17:42:45 poseidon systemd[1]: matrix-mautrix-signal.service: Main process exited, code=exited, status=15/n/a


```

How do I get past this?  

Other notes / tips / FYI's:

- I had been using Signal just fine with Matrix up until a few weeks ago when I sent a message into a group chat and it stopped syncing that room.  
- Other rooms were still going.  
- I haven't updated Ansible/Matrix/Signal since October 2023

**Pull request merged**